packages/server/src/common/command/command.ts

```diff
@@ -108,12 +110,14 @@ export class CompoundCommand implements Command {
         const alreadyRedone: Command[] = [];
 
         try {
-            this.commands.forEach(command => {
+            for (const command of this.commands) {
                 command.redo();
                 alreadyRedone.unshift(command);
-            });
+            }
         } catch (err) {
-            alreadyRedone.forEach(command => command.undo());
+            for (const command of alreadyRedone) {
+                command.undo();
```

Comment:
Should we not also await the `undo` here to make sure that the order is properly kept and there are no problems?